src/runtime_src/core/tools/xbflash2/OO_Program_Qspips.cpp

```diff
@@ -0,0 +1,295 @@
+/**
+ * Copyright (C) 2022 Licensed under the Apache License, Version
+ * 2.0 (the "License"). You may not use this file except in
+ * compliance with the License. A copy of the License is located
+ * at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "OO_Program_Qspips.h"
+#include "tools/common/XBUtilitiesCore.h"
+
+#include "core/pcie/tools/xbflash.qspi/firmware_image.h"
+#include "core/pcie/tools/xbflash.qspi/pcidev.h"
+#include "core/pcie/tools/xbflash.qspi/xqspips.h"
+
+namespace XBU = XBUtilities;
+
+// 3rd Party Library - Include Files
+#include <boost/program_options.hpp>
+#include <boost/format.hpp>
+#include <boost/algorithm/string.hpp>
+namespace po = boost::program_options;
+
+// System - Include Files
+#include <iostream>
+#include <fstream>
+
+namespace {
+
+int 
```

Comment:
Please make this a void.  There is no need to return the status of this function for a given operation.  If there is an issue, use exceptions.